### PR TITLE
Pin c free

### PR DESCRIPTION
**Pull request by taoli4rs**
Allow free pin location assignment. In this mode:
1. If user provide a sdc with set_pin_loc, it will be ignored;
2. If user does not provide a sdc with pin location constraint, pin_c will not create any pin constraint for it;
3. VPR is allowed to assign pin freely.

Usage:
...
**pin_loc_assign_method free**
place
...

### CODE REVIEW

**Comment by ravic-rs** [comment]
@taoli4rs we recently made changes to the files that shows up as conflicts, try merging main into your branches.

File: src/NewProject/add_constraints_form.ui [hunk 1]
```diff
@@ -1,129 +1,136 @@
...
-        <widget class="QGroupBox" name="groupBox">
-         <property name="title">
-          <string>Pin Location Assign Method:</string>
-         </property>
-         <layout class="QFormLayout" name="formLayout">
-          <item row="0" column="0">
-           <widget class="QRadioButton" name="select_random">
-            <property name="text">
-             <string>Random</string>
+    <class>addConstraintsForm</class>
```
Comment (KochynVolodymyr) [inline src/NewProject/add_constraints_form.ui:3]: Something wrong with UI file:
image

It should be simple change

**Comment by alain-rs** [comment]
@taoli4rs , please update your branch there are conflicts.
Remove your change in the GUI form. The Tcl command is the most important.

**Comment by taoli4rs** [comment]
Close this PR. A PR for command only is created: https://github.com/os-fpga/FOEDAG/pull/716

**Pull request closed without merging**